phoenix-core-client/src/main/java/org/apache/phoenix/optimize/QueryOptimizer.java

```diff
@@ -283,13 +283,17 @@ private List<QueryPlan> getApplicablePlansForSingleFlatQuery(QueryPlan dataPlan,
             plans.add(dataPlan);
             hintedPlan = getHintedQueryPlan(statement, translatedIndexSelect, indexes,
                     targetColumns, parallelIteratorFactory, plans);
-            if (hintedPlan != null) {
+            if (hintedPlan != null ) {
                 PTable index = hintedPlan.getTableRef().getTable();
-                if (stopAtBestPlan && hintedPlan.isApplicable() && (index.getIndexWhere() == null
-                        || isPartialIndexUsable(select, dataPlan, index))) {
-                    return Collections.singletonList(hintedPlan);
+                // Ignore any index hint with a CDC index
```

Comment:
As it is a truncated index, it only includes index rows for the data table rows that have changed within the max lookback window.